### ISSUE TITLE
The storage table directories still exists even if the iceberg materialized view is droped

### ISSUE THREAD

**Issue by hseagle** [trinodb/trino#17765]
After the iceberg materialized view is dropped, the data directories of the storage table still exists.  According to the trino documents, the storage table and it's data should be dropped also.  

It can be reproduced like this. smoke_ice_person is a iceberg table

1.  create materialized view mv_person_stats as select count(*) from smoke_ice_person;
2. show create materialized view mv_person_stats, the storage table is demo.st_7671ab7da83c413b9a649c6014c7dbce
3. refresh materialized view mv_person_stats
4. drop materialized view mv_person_stats

In the last step, the materialized view mv_person_stats is dropped successfully, but the data directory st_7671ab7da83c413b9a649c6014c7dbce still exists.

The trino version number is Trino 418.

**Comment by hseagle** [comment]
After debugging the dropTable func in ThriftHiveMetaStore,  I found that the storage table is treated as external table, that's why the data directory is not removed.

**Comment by findinpath** [comment]
tldr; I can't seem to reproduce the issue in Trino `446`.

```
testing/bin/ptl env up --environment singlenode-spark-iceberg
```

```
trino> select version();
      _col0      
-----------------
 446-74-g5d4d48b 

trino> create table iceberg.default.t1 (c integer);
CREATE TABLE
trino> insert into iceberg.default.t1 values 1;
INSERT: 1 row

trino> insert into iceberg.default.t1 values 2;
INSERT: 1 row

trino> insert into iceberg.default.t1 values 3;
INSERT: 1 row

trino> create materialized view iceberg.default.mv_t1 as select count(*)  as cnt from iceberg.default.t1;
CREATE MATERIALIZED VIEW
trino> show create materialized view iceberg.default.mv_t1;
                                       Create Materialized View                                        
-------------------------------------------------------------------------------------------------------
 CREATE MATERIALIZED VIEW iceberg.default.mv_t1                                                        
 WITH (                                                                                                
    format = 'PARQUET',                                                                                
    format_version = 2,                                                                                
    location = 'hdfs://hadoop-master:9000/user/hive/warehouse/mv_t1-8976fca97dee42cb8583046646de1445', 
    storage_schema = 'default'                                                                         
 ) AS                                                                                                  
 SELECT count(*) cnt                                                                                   
 FROM                                                                                                  
   iceberg.default.t1                                                                                  
(1 row)

trino> refresh materialized view iceberg.default.mv_t1;
REFRESH MATERIALIZED VIEW: 1 row
```

```
➜  ~ docker container exec -ti ptl-hadoop-master /bin/bash
[root@hadoop-master /]# hdfs dfs -ls -R /user/hive/warehouse
drwxr-xr-x   - hive supergroup          0 2021-01-18 15:37 /user/hive/warehouse/information_schema.db
drwxrwxrwx   - hive supergroup          0 2024-05-08 06:18 /user/hive/warehouse/mv_t1-8976fca97dee42cb8583046646de1445
drwxrwxrwx   - hive supergroup          0 2024-05-08 06:18 /user/hive/warehouse/mv_t1-8976fca97dee42cb8583046646de1445/data
-rw-r--r--   3 hive supergroup        231 2024-05-08 06:18 /user/hive/warehouse/mv_t1-8976fca97dee42cb8583046646de1445/data/20240508_061856_00009_ayjfd-672afaf5-e1b9-490a-b76f-6cd4ec8af5b5.parquet
drwxrwxrwx   - hive supergroup          0 2024-05-08 06:18 /user/hive/warehouse/mv_t1-8976fca97dee42cb8583046646de1445/metadata
-rw-r--r--   3 hive supergroup        990 2024-05-08 06:18 /user/hive/warehouse/mv_t1-8976fca97dee42cb8583046646de1445/metadata/00000-a54e6145-3e70-4978-bdee-2a3b098feef2.metadata.json
-rw-r--r--   3 hive supergroup       2996 2024-05-08 06:18 /user/hive/warehouse/mv_t1-8976fca97dee42cb8583046646de1445/metadata/00001-9d183ed0-4dce-43a0-9df7-f5e520e183ac.metadata.json
-rw-r--r--   3 hive supergroup       6657 2024-05-08 06:18 /user/hive/warehouse/mv_t1-8976fca97dee42cb8583046646de1445/metadata/255af019-4a38-4e72-8054-3c25dab63b7a-m0.avro
-rw-r--r--   3 hive supergroup       4278 2024-05-08 06:18 /user/hive/warehouse/mv_t1-8976fca97dee42cb8583046646de1445/metadata/snap-8619271560896524523-1-255af019-4a38-4e72-8054-3c25dab63b7a.avro
-rw-r--r--   3 hive supergroup       4095 2024-05-08 06:18 /user/hive/warehouse/mv_t1-8976fca97dee42cb8583046646de1445/metadata/snap-8669262783964620424-1-4de68edc-21cb-46f0-ba78-e85c71deb9a7.avro
.....
drwxrwxrwx   - hive supergroup          0 2024-05-08 06:17 /user/hive/warehouse/t1-2e93ccadebfe40ebb341e6a7513f3ef0
drwxrwxrwx   - hive supergroup          0 2024-05-08 06:17 /user/hive/warehouse/t1-2e93ccadebfe40ebb341e6a7513f3ef0/data
-rw-r--r--   3 hive supergroup        207 2024-05-08 06:17 /user/hive/warehouse/t1-2e93ccadebfe40ebb341e6a7513f3ef0/data/20240508_061659_00002_ayjfd-20f7a4aa-b90e-45d6-ada5-5c1806fb9686.parquet
-rw-r--r--   3 hive supergroup        207 2024-05-08 06:17 /user/hive/warehouse/t1-2e93ccadebfe40ebb341e6a7513f3ef0/data/20240508_061705_00003_ayjfd-09cf2d86-d71f-4984-9324-2c6ab33534a5.parquet
-rw-r--r--   3 hive supergroup        207 2024-05-08 06:17 /user/hive/warehouse/t1-2e93ccadebfe40ebb341e6a7513f3ef0/data/20240508_061709_00004_ayjfd-3de63d84-5378-47fa-830c-ee9f1d266b1a.parquet
drwxrwxrwx   - hive supergroup          0 2024-05-08 06:17 /user/hive/warehouse/t1-2e93ccadebfe40ebb341e6a7513f3ef0/metadata
-rw-r--r--   3 hive supergroup       1840 2024-05-08 06:16 /user/hive/warehouse/t1-2e93ccadebfe40ebb341e6a7513f3ef0/metadata/00000-7ed6bb31-dbd3-4b1f-bbc7-c5bbf09c3ef1.metadata.json
-rw-r--r--   3 hive supergroup       2950 2024-05-08 06:17 /user/hive/warehouse/t1-2e93ccadebfe40ebb341e6a7513f3ef0/metadata/00001-04395cf3-00e1-4710-a91b-0ff80b8bbfdc.metadata.json
-rw-r--r--   3 hive supergroup       3700 2024-05-08 06:17 /user/hive/warehouse/t1-2e93ccadebfe40ebb341e6a7513f3ef0/metadata/00002-47929a9f-c155-413e-844e-0d9bc07399d8.metadata.json
-rw-r--r--   3 hive supergroup       4811 2024-05-08 06:17 /user/hive/warehouse/t1-2e93ccadebfe40ebb341e6a7513f3ef0/metadata/00003-c5b031fb-ccf9-40b7-8e8a-adf721cdc465.metadata.json
-rw-r--r--   3 hive supergroup       5562 2024-05-08 06:17 /user/hive/warehouse/t1-2e93ccadebfe40ebb341e6a7513f3ef0/metadata/00004-83eb4ca6-9cfc-4903-bead-832702b4d626.metadata.json
-rw-r--r--   3 hive supergroup       6673 2024-05-08 06:17 /user/hive/warehouse/t1-2e93ccadebfe40ebb341e6a7513f3ef0/metadata/00005-1bc8370c-5778-4eba-9b04-d56c18dd6891.metadata.json
-rw-r--r--   3 hive supergroup       7424 2024-05-08 06:17 /user/hive/warehouse/t1-2e93ccadebfe40ebb341e6a7513f3ef0/metadata/00006-718fd4c5-2707-4bcc-b130-5d48632e22c5.metadata.json
-rw-r--r--   3 hive supergroup        302 2024-05-08 06:17 /user/hive/warehouse/t1-2e93ccadebfe40ebb341e6a7513f3ef0/metadata/20240508_061659_00002_ayjfd-cd518eea-33e0-403b-ab29-195d48633236.stats
-rw-r--r--   3 hive supergroup        318 2024-05-08 06:17 /user/hive/warehouse/t1-2e93ccadebfe40ebb341e6a7513f3ef0/metadata/20240508_061705_00003_ayjfd-9f73f7d4-4168-4375-9ac2-f57d0931d5ab.stats
-rw-r--r--   3 hive supergroup        326 2024-05-08 06:17 /user/hive/warehouse/t1-2e93ccadebfe40ebb341e6a7513f3ef0/metadata/20240508_061709_00004_ayjfd-a0521c25-5503-427a-b3d3-580b2e5af4ed.stats
-rw-r--r--   3 hive supergroup       6649 2024-05-08 06:17 /user/hive/warehouse/t1-2e93ccadebfe40ebb341e6a7513f3ef0/metadata/97760bad-b83e-4f70-9ce7-841b19b3b175-m0.avro
-rw-r--r--   3 hive supergroup       6648 2024-05-08 06:17 /user/hive/warehouse/t1-2e93ccadebfe40ebb341e6a7513f3ef0/metadata/b249a755-279c-4fc3-838d-fe44bac350f0-m0.avro
-rw-r--r--   3 hive supergroup       6647 2024-05-08 06:17 /user/hive/warehouse/t1-2e93ccadebfe40ebb341e6a7513f3ef0/metadata/e693d02a-d44f-4b8d-b11a-31d5e995b309-m0.avro
-rw-r--r--   3 hive supergroup       4095 2024-05-08 06:16 /user/hive/warehouse/t1-2e93ccadebfe40ebb341e6a7513f3ef0/metadata/snap-3039712757649811570-1-ab5e4728-2e64-4215-ab86-ca843b1ecfa9.avro
-rw-r--r--   3 hive supergroup       4273 2024-05-08 06:17 /user/hive/warehouse/t1-2e93ccadebfe40ebb341e6a7513f3ef0/metadata/snap-4205116603863994627-1-b249a755-279c-4fc3-838d-fe44bac350f0.avro
-rw-r--r--   3 hive supergroup       4334 2024-05-08 06:17 /user/hive/warehouse/t1-2e93ccadebfe40ebb341e6a7513f3ef0/metadata/snap-4785004897947812895-1-97760bad-b83e-4f70-9ce7-841b19b3b175.avro
-rw-r--r--   3 hive supergroup       4379 2024-05-08 06:17 /user/hive/warehouse/t1-2e93ccadebfe40ebb341e6a7513f3ef0/metadata/snap-5510061461970578438-1-e693d02a-d44f-4b8d-b11a-31d5e995b309.avro
```

```
trino> drop materialized view iceberg.default.mv_t1;
DROP MATERIALIZED VIEW
```

```
[root@hadoop-master /]# hdfs dfs -ls -R /user/hive/warehouse
drwxr-xr-x   - hive supergroup          0 2021-01-18 15:37 /user/hive/warehouse/information_schema.db
drwxr-xr-x   - hive supergroup          0 2021-01-18 15:37 /user/hive/warehouse/sys.db
drwxr-xr-x   - hive supergroup          0 2021-01-18 15:37 /user/hive/warehouse/sys.db/bucketing_cols
drwxr-xr-x   - hive supergroup          0 2021-01-18 15:37 /user/hive/warehouse/sys.db/cds
drwxr-xr-x   - hive supergroup          0 2021-01-18 15:37 /user/hive/warehouse/sys.db/columns_v2
drwxr-xr-x   - hive supergroup          0 2021-01-18 15:37 /user/hive/warehouse/sys.db/database_params
drwxr-xr-x   - hive supergroup          0 2021-01-18 15:37 /user/hive/warehouse/sys.db/db_privs
drwxr-xr-x   - hive supergroup          0 2021-01-18 15:37 /user/hive/warehouse/sys.db/db_version
drwxr-xr-x   - hive supergroup          0 2021-01-18 15:37 /user/hive/warehouse/sys.db/dbs
drwxr-xr-x   - hive supergroup          0 2021-01-18 15:37 /user/hive/warehouse/sys.db/funcs
drwxr-xr-x   - hive supergroup          0 2021-01-18 15:37 /user/hive/warehouse/sys.db/global_privs
drwxr-xr-x   - hive supergroup          0 2021-01-18 15:37 /user/hive/warehouse/sys.db/key_constraints
drwxr-xr-x   - hive supergroup          0 2021-01-18 15:37 /user/hive/warehouse/sys.db/mv_creation_metadata
drwxr-xr-x   - hive supergroup          0 2021-01-18 15:37 /user/hive/warehouse/sys.db/mv_tables_used
drwxr-xr-x   - hive supergroup          0 2021-01-18 15:37 /user/hive/warehouse/sys.db/part_col_privs
drwxr-xr-x   - hive supergroup          0 2021-01-18 15:37 /user/hive/warehouse/sys.db/part_col_stats
drwxr-xr-x   - hive supergroup          0 2021-01-18 15:37 /user/hive/warehouse/sys.db/part_privs
drwxr-xr-x   - hive supergroup          0 2021-01-18 15:37 /user/hive/warehouse/sys.db/partition_key_vals
drwxr-xr-x   - hive supergroup          0 2021-01-18 15:37 /user/hive/warehouse/sys.db/partition_keys
drwxr-xr-x   - hive supergroup          0 2021-01-18 15:37 /user/hive/warehouse/sys.db/partition_params
drwxr-xr-x   - hive supergroup          0 2021-01-18 15:37 /user/hive/warehouse/sys.db/partitions
drwxr-xr-x   - hive supergroup          0 2021-01-18 15:37 /user/hive/warehouse/sys.db/role_map
drwxr-xr-x   - hive supergroup          0 2021-01-18 15:37 /user/hive/warehouse/sys.db/roles
drwxr-xr-x   - hive supergroup          0 2021-01-18 15:37 /user/hive/warehouse/sys.db/sd_params
drwxr-xr-x   - hive supergroup          0 2021-01-18 15:37 /user/hive/warehouse/sys.db/sds
drwxr-xr-x   - hive supergroup          0 2021-01-18 15:37 /user/hive/warehouse/sys.db/sequence_table
drwxr-xr-x   - hive supergroup          0 2021-01-18 15:37 /user/hive/warehouse/sys.db/serde_params
drwxr-xr-x   - hive supergroup          0 2021-01-18 15:37 /user/hive/warehouse/sys.db/serdes
drwxr-xr-x   - hive supergroup          0 2021-01-18 15:37 /user/hive/warehouse/sys.db/skewed_col_names
drwxr-xr-x   - hive supergroup          0 2021-01-18 15:37 /user/hive/warehouse/sys.db/skewed_col_value_loc_map
drwxr-xr-x   - hive supergroup          0 2021-01-18 15:37 /user/hive/warehouse/sys.db/skewed_string_list
drwxr-xr-x   - hive supergroup          0 2021-01-18 15:37 /user/hive/warehouse/sys.db/skewed_string_list_values
drwxr-xr-x   - hive supergroup          0 2021-01-18 15:37 /user/hive/warehouse/sys.db/skewed_values
drwxr-xr-x   - hive supergroup          0 2021-01-18 15:37 /user/hive/warehouse/sys.db/sort_cols
drwxr-xr-x   - hive supergroup          0 2021-01-18 15:37 /user/hive/warehouse/sys.db/tab_col_stats
drwxr-xr-x   - hive supergroup          0 2021-01-18 15:37 /user/hive/warehouse/sys.db/table_params
drwxr-xr-x   - hive supergroup          0 2021-01-18 15:37 /user/hive/warehouse/sys.db/tbl_col_privs
drwxr-xr-x   - hive supergroup          0 2021-01-18 15:37 /user/hive/warehouse/sys.db/tbl_privs
drwxr-xr-x   - hive supergroup          0 2021-01-18 15:37 /user/hive/warehouse/sys.db/tbls
drwxr-xr-x   - hive supergroup          0 2021-01-18 15:37 /user/hive/warehouse/sys.db/wm_mappings
drwxr-xr-x   - hive supergroup          0 2021-01-18 15:37 /user/hive/warehouse/sys.db/wm_pools
drwxr-xr-x   - hive supergroup          0 2021-01-18 15:37 /user/hive/warehouse/sys.db/wm_pools_to_triggers
drwxr-xr-x   - hive supergroup          0 2021-01-18 15:37 /user/hive/warehouse/sys.db/wm_resourceplans
drwxr-xr-x   - hive supergroup          0 2021-01-18 15:37 /user/hive/warehouse/sys.db/wm_triggers
drwxrwxrwx   - hive supergroup          0 2024-05-08 06:17 /user/hive/warehouse/t1-2e93ccadebfe40ebb341e6a7513f3ef0
drwxrwxrwx   - hive supergroup          0 2024-05-08 06:17 /user/hive/warehouse/t1-2e93ccadebfe40ebb341e6a7513f3ef0/data
-rw-r--r--   3 hive supergroup        207 2024-05-08 06:17 /user/hive/warehouse/t1-2e93ccadebfe40ebb341e6a7513f3ef0/data/20240508_061659_00002_ayjfd-20f7a4aa-b90e-45d6-ada5-5c1806fb9686.parquet
-rw-r--r--   3 hive supergroup        207 2024-05-08 06:17 /user/hive/warehouse/t1-2e93ccadebfe40ebb341e6a7513f3ef0/data/20240508_061705_00003_ayjfd-09cf2d86-d71f-4984-9324-2c6ab33534a5.parquet
-rw-r--r--   3 hive supergroup        207 2024-05-08 06:17 /user/hive/warehouse/t1-2e93ccadebfe40ebb341e6a7513f3ef0/data/20240508_061709_00004_ayjfd-3de63d84-5378-47fa-830c-ee9f1d266b1a.parquet
drwxrwxrwx   - hive supergroup          0 2024-05-08 06:17 /user/hive/warehouse/t1-2e93ccadebfe40ebb341e6a7513f3ef0/metadata
-rw-r--r--   3 hive supergroup       1840 2024-05-08 06:16 /user/hive/warehouse/t1-2e93ccadebfe40ebb341e6a7513f3ef0/metadata/00000-7ed6bb31-dbd3-4b1f-bbc7-c5bbf09c3ef1.metadata.json
-rw-r--r--   3 hive supergroup       2950 2024-05-08 06:17 /user/hive/warehouse/t1-2e93ccadebfe40ebb341e6a7513f3ef0/metadata/00001-04395cf3-00e1-4710-a91b-0ff80b8bbfdc.metadata.json
-rw-r--r--   3 hive supergroup       3700 2024-05-08 06:17 /user/hive/warehouse/t1-2e93ccadebfe40ebb341e6a7513f3ef0/metadata/00002-47929a9f-c155-413e-844e-0d9bc07399d8.metadata.json
-rw-r--r--   3 hive supergroup       4811 2024-05-08 06:17 /user/hive/warehouse/t1-2e93ccadebfe40ebb341e6a7513f3ef0/metadata/00003-c5b031fb-ccf9-40b7-8e8a-adf721cdc465.metadata.json
-rw-r--r--   3 hive supergroup       5562 2024-05-08 06:17 /user/hive/warehouse/t1-2e93ccadebfe40ebb341e6a7513f3ef0/metadata/00004-83eb4ca6-9cfc-4903-bead-832702b4d626.metadata.json
-rw-r--r--   3 hive supergroup       6673 2024-05-08 06:17 /user/hive/warehouse/t1-2e93ccadebfe40ebb341e6a7513f3ef0/metadata/00005-1bc8370c-5778-4eba-9b04-d56c18dd6891.metadata.json
-rw-r--r--   3 hive supergroup       7424 2024-05-08 06:17 /user/hive/warehouse/t1-2e93ccadebfe40ebb341e6a7513f3ef0/metadata/00006-718fd4c5-2707-4bcc-b130-5d48632e22c5.metadata.json
-rw-r--r--   3 hive supergroup        302 2024-05-08 06:17 /user/hive/warehouse/t1-2e93ccadebfe40ebb341e6a7513f3ef0/metadata/20240508_061659_00002_ayjfd-cd518eea-33e0-403b-ab29-195d48633236.stats
-rw-r--r--   3 hive supergroup        318 2024-05-08 06:17 /user/hive/warehouse/t1-2e93ccadebfe40ebb341e6a7513f3ef0/metadata/20240508_061705_00003_ayjfd-9f73f7d4-4168-4375-9ac2-f57d0931d5ab.stats
-rw-r--r--   3 hive supergroup        326 2024-05-08 06:17 /user/hive/warehouse/t1-2e93ccadebfe40ebb341e6a7513f3ef0/metadata/20240508_061709_00004_ayjfd-a0521c25-5503-427a-b3d3-580b2e5af4ed.stats
-rw-r--r--   3 hive supergroup       6649 2024-05-08 06:17 /user/hive/warehouse/t1-2e93ccadebfe40ebb341e6a7513f3ef0/metadata/97760bad-b83e-4f70-9ce7-841b19b3b175-m0.avro
-rw-r--r--   3 hive supergroup       6648 2024-05-08 06:17 /user/hive/warehouse/t1-2e93ccadebfe40ebb341e6a7513f3ef0/metadata/b249a755-279c-4fc3-838d-fe44bac350f0-m0.avro
-rw-r--r--   3 hive supergroup       6647 2024-05-08 06:17 /user/hive/warehouse/t1-2e93ccadebfe40ebb341e6a7513f3ef0/metadata/e693d02a-d44f-4b8d-b11a-31d5e995b309-m0.avro
-rw-r--r--   3 hive supergroup       4095 2024-05-08 06:16 /user/hive/warehouse/t1-2e93ccadebfe40ebb341e6a7513f3ef0/metadata/snap-3039712757649811570-1-ab5e4728-2e64-4215-ab86-ca843b1ecfa9.avro
-rw-r--r--   3 hive supergroup       4273 2024-05-08 06:17 /user/hive/warehouse/t1-2e93ccadebfe40ebb341e6a7513f3ef0/metadata/snap-4205116603863994627-1-b249a755-279c-4fc3-838d-fe44bac350f0.avro
-rw-r--r--   3 hive supergroup       4334 2024-05-08 06:17 /user/hive/warehouse/t1-2e93ccadebfe40ebb341e6a7513f3ef0/metadata/snap-4785004897947812895-1-97760bad-b83e-4f70-9ce7-841b19b3b175.avro
-rw-r--r--   3 hive supergroup       4379 2024-05-08 06:17 /user/hive/warehouse/t1-2e93ccadebfe40ebb341e6a7513f3ef0/metadata/snap-5510061461970578438-1-e693d02a-d44f-4b8d-b11a-31d5e995b309.avro
```

The data directory of the storage table is gone.

**Comment by findepi** [comment]
> tldr; I can't seem to reproduce the issue in Trino `446`.

**Comment by alexjo2144** [comment]
Probably fixed by: https://github.com/trinodb/trino/pull/20892
